Backend/DataAccessLayer/Repositories/UserManagementRepository.cs

```diff
@@ -11,85 +11,56 @@ namespace DataAccessLayer.Repositories
 {
     public class UserManagementRepository
     {
-        public int CreateNewUser(User user)
+        public User CreateNewUser(User user, DatabaseContext _db)
         {
-            using (var _db = new DatabaseContext())
-            {
-                try
-                {
-                    _db.Users.Add(user);
-                    return _db.SaveChanges();
-                }
-                catch (Exception)
-                {
-                    return 0;
-                }
-            }
+            _db.Entry(user).State = EntityState.Added;

```

Comment:
This is interesting syntax. Is this a different way to add a user to the DB?

Reply:
.Add sets the sate to added, this gives a bit more control over what state you want to set the entity.